requirements/constraints.txt

```diff
@@ -34,7 +34,10 @@ django-simple-history==3.4.0
 
 # constrained in opaque_keys. migration guide here: https://pymongo.readthedocs.io/en/4.0/migrate-to-pymongo4.html
 # Major upgrade will be done in separate ticket.
-pymongo<4.0.0
+pymongo<4.4.1
+
+# To override the constraint of edx-lint
```

Comment:
```suggestion
# To override the constraint of edx-lint
# This can be removed once https://github.com/openedx/edx-platform/issues/34586 is resolved 
# and the upstream constraint in edx-lint has been removed.
```